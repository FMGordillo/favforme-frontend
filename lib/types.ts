/**
 * TODO: This should be aligned with the schema.prisma (maybe monolithic app?)
 */

export type SocialNetworkName =
  | "FACEBOOK"
  | "TWITTER"
  | "INSTAGRAM"
  | "LINKEDIN";

export type Status = "DRAFT" | "INPROGRESS" | "COMPLETED" | "CANCELLED";

export interface SocialNetwork {
  id: number;
  type: SocialNetworkName;
  name?: string; // "others"
  link: string;
}

export interface Organization {
  id: number;
  name: string;
  logo?: string;
  history?: string;
  homepage?: string;
  actions: ActionI[];
  socialNetworks: SocialNetwork[];
}

export interface ActionI {
  id: number;
  title: string;
  description: string;
  peopleBeneficted?: number;
  current: number; // $$$
  objective: number; // $$$
  organization: Organization;
  organizationId: number;
  deleted: boolean;
  status: Status;
}

export interface User {
  id: string;
  email: string;
}
/**
 * TODO: Move this to Apollo requests
 */
export interface GetActionsData {
  actions: ActionI[];
}
