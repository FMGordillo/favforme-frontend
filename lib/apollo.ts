import { ApolloClient, InMemoryCache } from "@apollo/client";

const client = new ApolloClient({
  ssrMode: typeof window === "undefined",
  uri: `${process.env.NEXT_PUBLIC_BACKEND_URL}/graphql`,
  cache: new InMemoryCache(),
});

export default client;
