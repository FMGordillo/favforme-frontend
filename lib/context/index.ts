export * from "./ModalContext";
export * from "./NotificationContext";
