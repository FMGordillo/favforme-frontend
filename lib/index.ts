export const toPascalCase = (string: string): string =>
  string.replace(/(\w)(\w*)/g, function (g0, g1, g2) {
    return g1.toUpperCase() + g2.toLowerCase();
  });
