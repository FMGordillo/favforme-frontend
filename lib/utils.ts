/**
 * @default 0
 */
export const toPascalCase = (string: string | undefined): string =>
  string
    ? string.replace(/(\w)(\w*)/g, function (_g0, g1, g2) {
        return g1.toUpperCase() + g2.toLowerCase();
      })
    : "";
