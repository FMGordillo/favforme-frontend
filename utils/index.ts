// export const isNotProd = process.env.NEXT_PUBLIC_ENVIRONMENT !== "production";
export const isNotProd = true;
