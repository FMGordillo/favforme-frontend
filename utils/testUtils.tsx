import { render } from "@testing-library/react";
import { ReactChild } from "react";
import { ThemeProvider } from "styled-components";
import { SWRConfig } from "swr";
import { fetcher } from "../lib/queries";
import { lightTheme } from "./styled";

const Providers = ({ children }: { children: ReactChild }) => {
  return (
    <SWRConfig
      value={{
        refreshInterval: 10000,
        fetcher,
      }}
    >
      <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
    </SWRConfig>
  );
};

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const customRender = (ui: JSX.Element, options = {}) =>
  // @ts-ignore
  render(ui, { wrapper: Providers, ...options });

import "@testing-library/jest-dom";
export * from "@testing-library/react";
export { customRender as render };
