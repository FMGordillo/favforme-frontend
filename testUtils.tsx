import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "./utils/styled";
// import { ThemeProvider } from "my-ui-lib"
// import { TranslationProvider } from "my-i18n-lib"
// import defaultStrings from "i18n/en-x-default"

const Providers = ({ children }) => {
  // return children;
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
