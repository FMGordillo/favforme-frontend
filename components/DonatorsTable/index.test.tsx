import { DonationI } from "@/lib/types";
import { DonatorsTable } from "./";
import { render } from "@/utils/testUtils";

const data: DonationI[] = [
  {
    id: "1",
    currency: "ARS",
    user: {
      id: "2",
      name: null,
      surname: null,
      email: "test@tester.com",
      userType: "USER",
    },
    action: {
      id: "3",
      title: "test",
      description: "A test description",
      objective: "5",
      current: "1",
      createdAt: new Date().toString(),
      status: "INPROGRESS",
      organization: {
        id: "4",
        name: "test",
        actions: [],
        socialNetworks: [],
      },
      deleted: false,
      ods: [],
    },
    actionId: "3",
    userId: "2",
    amount: "100",
    paymentStatus: "SUCCESS",
    createdAt: new Date().toString(),
    updatedAt: new Date().toString(),
  },
];

describe("[Container] About us", () => {
  it("renders correctly", async () => {
    const { asFragment } = render(<DonatorsTable data={data} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
