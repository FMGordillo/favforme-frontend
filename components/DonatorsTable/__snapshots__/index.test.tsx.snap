// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Container] About us renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 100%;
  max-width: 1024px;
  margin-bottom: 1em;
  text-align: center;
  border-collapse: collapse;
  border-bottom: 3px solid #4f6be6;
}

.c1 {
  color: white;
  background-color: #4f6be6;
}

.c1 tr > td {
  padding: 8px;
}

.c2 {
  border-left: 3px solid #4f6be6;
  border-right: 3px solid #4f6be6;
}

.c2:first-of-type {
  border-left-width: 6px;
}

.c2:last-of-type {
  border-right-width: 6px;
}

@media (max-width:600px) {
  .c2:nth-child(2) {
    display: none;
  }
}

<table
    class="c0"
  >
    <thead
      class="c1"
    >
      <tr
        class=""
      >
        <td
          class="c2"
        >
          Fecha
        </td>
        <td
          class="c2"
        >
          Empresa / particular
        </td>
        <td
          class="c2"
        >
          Contribuidor
        </td>
        <td
          class="c2"
        >
          Total (pesos argentinos)
        </td>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class="c2"
        >
          2012/12/12
        </td>
        <td
          class="c2"
        >
          Donante particular
        </td>
        <td
          class="c2"
        >
          Anonimo/a
        </td>
        <td
          class="c2"
        >
          $100
        </td>
      </tr>
    </tbody>
  </table>
  .c0 {
  border: none;
  cursor: pointer;
  font-size: 1.5rem;
  padding: 12px 28px;
  border-radius: 64px;
  font-family: abel,sans-serif;
  -webkit-transition: background-color 300ms ease-out,color 300ms ease-out;
  transition: background-color 300ms ease-out,color 300ms ease-out;
  background-color: #4f6be6;
}

.c0 > a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: #9BAAF6;
}

@media (max-width:600px) {

}

<button
    class="c0"
    style="display: none;"
  >
    Cargar mas
  </button>
</DocumentFragment>
`;
